libs/SmartSync/SmartSync/Classes/Util/SFSyncOptions.h

```diff
@@ -40,6 +44,11 @@ extern NSString * const kSFSyncOptionsMergeMode;
 + (SFSyncOptions*) newSyncOptionsForSyncUp:(NSArray*)fieldlist;
 + (SFSyncOptions*) newSyncOptionsForSyncUp:(NSArray*)fieldlist mergeMode:(SFSyncStateMergeMode)mergeMode;
 
++ (SFSyncOptions*) newSyncOptionsForSyncUp:(NSArray*)fieldlist
```

Comment:
If you have all three then field list won't be used.
I have to keep field list to be backwards compatible and because in most cases it works for both create and update.
I couldn't decide if I should single out creates (by adding only createFieldlist) or udpates (by adding updateFieldlist) so I ended up adding both.